app/processing/settlement/get-last-settlement.js

```diff
@@ -0,0 +1,31 @@
+const db = require('../../data')
+
+const getLastSettlement = async (settlement, transaction) => {
+  return db.settlement.findOne({
+    transaction,
+    attributes: [
+      'paymentRequestId',
+      'invoiceNumber',
+      'reference',
+      'settled',
+      'settlementDate',
+      'value'
+    ],
+    where: {
+      settlementDate: {
+        [db.Sequelize.Op.lt]: settlement.settlementDate
```

Comment:
Make sure settlement date is earlier than current settlement so we don't get either the same one again or a later one that can't be part of the current payment we're generating the statement for.